prereise/gather/demanddata/transportation_electrification/charging.py

```diff
@@ -0,0 +1,114 @@
+def get_charging_power(power, trip_strategy, location_strategy, kwh, trip_data):
+    """Determines the charging power. Returns either param power or 0.
+
+    :param int power: charger power, EVSE kW.
+    :param int trip_strategy: a flag that determines which trips to consider, 1-anytrip number, 2-last trip.
+    :param int location_strategy: where the vehicle can charge-1, 2, 3, 4, or 5; 
+        1-home only, 2-home and work related, 3-anywhere if possibile, 
+        4-home and school only, 5-home and work and school.
+    :param float kwh: kwhmi * veh_range, amount of energy needed to charge vehicle.
```

Comment:
Yes, so if the battery SOC at the end of a trip is less than this kwh value, then it'll indicate that the vehicle needs to be charged.

Reply:
Doesn't the vehicle always need to be charged at least a little bit at the end of a every trip? Even if there are zero-energy/zero-mileage trips, won't that end up getting handled downstream because the charging time will be zero?